src/main/java/org/helpberkeley/memberdata/DriverPostFormat.java

```diff
@@ -76,20 +77,16 @@ private static DriverPostFormat doCreate(ApiClient apiClient,
         // FIX THIS, DS: check warnings here?
 
         DriverPostFormat driverPostFormat;
-
-        switch (controlBlock.getVersion()) {
-            case Constants.CONTROL_BLOCK_VERSION_UNKNOWN:
-                throw new MemberDataException("Control block not found");
-            case Constants.CONTROL_BLOCK_VERSION_200:
-                driverPostFormat = new DriverPostFormatV200();
-                break;
-            case Constants.CONTROL_BLOCK_VERSION_300:
-                driverPostFormat = new DriverPostFormatV300();
-                break;
-            case Constants.CONTROL_BLOCK_VERSION_1:
-            default:
-                throw new MemberDataException(
-                        "Control block version " + controlBlock.getVersion() + " is not supported.\n");
+        String version = controlBlock.getVersion();
+        if (version.equals(Constants.CONTROL_BLOCK_VERSION_UNKNOWN)) {
```

Comment:
Either remove the version variable and call contolBlock.getVersion() at line 81, or use the version variable at line 89.